wire/msgtx.go

```diff
@@ -326,6 +413,13 @@ func (msg *MsgTx) Deserialize(r io.Reader) error {
 	return msg.BtcDecode(r, 0)
 }
 
+// DeserializeWitness decodes a transaction from r into the receiver, where the
+// encoded transaction format within r may possibly utilize the new serialization
+// format created to encode transactions bearing witness data within inputs.
+func (msg *MsgTx) DeserializeWitness(r io.Reader) error {
+	return msg.BtcDecode(r, WitnessVersion)
```

Comment:
First, note that this is an outdated diff. In the latest version `BtcDecode` has gained an additional parameter which controls the encoding type. 

Using the transaction version is insufficient, because this function might be deserializing the transaction encoded without the witness data because the transaction has been received from a non-upgraded peer. The server will never request the transaction from such a peer since it detects if it is witness enabled before sending the getdata message.